src/phases/execution/LookupCommands.yml

```diff
@@ -59,6 +59,16 @@ ForeignSmallDocTemplate:
   arr_key: {^Array: {of: *RandIntUpTo, number: 5}}
   data1: *Rand32BitInt
 
+# The document template for both local and foreign collections in NLJ Lookups. The doc contains
+# scalar, single element array, 8 elements array, simple object and nested object.
+LargeComplexDocTemplate:
+  int_key1: &RandInt0To1k {^RandomInt: {min: 0, max: 1000}}
+  int_key2: {^RandomInt: {min: 2000, max: 3000}}
+  arr1: [*RandInt0To1k]
+  arr8: {^Array: {of: *RandInt0To1k, number: 8}}
+  obj1: {obj: *RandInt0To1k}
+  obj4: {obj3: {obj2: {obj1: {obj: *RandInt0To1k}}}}
```

Comment:
[nit] recommend renaming 'obj' to something referencing that the value is an int. Here and above.